tango_simlib/tests/test_validator.py

```diff
@@ -0,0 +1,358 @@
+  - name: AsynchCmdReplyTimeout
+  - name: CentralLoggerEnabledDefault
+  - name: ConfigureTaskTimeout
+  - name: ControlModeDefault_A
+"""
+
+YAML_B = """
+class: DishMaster_B
+meta:
+  attributes:
+  - data_format: SCALAR_B
+    data_type: DevLong_B
+    description: Current  logging level to Central logging target
+    disp_level: OPERATOR
+    display_unit: No display unit
+    format: '%d'
+    label: loggingLevelCentral
+    max_alarm: Not specified
+    max_dim_x: 1
+    max_dim_y: 0
+    max_value: Not specified
+    min_alarm: Not specified
+    min_value: Not specified
+    name: loggingLevelCentral
+    standard_unit: No standard unit
+    unit: ''
+    writable: READ_WRITE
+    writable_attr_name: logginglevelcentral
+  - data_format: SCALAR
+    data_type: DevLong
+    description: Current  logging level to Element logging target
+    disp_level: OPERATOR
+    display_unit: No display unit
+    format: '%d'
+    label: loggingLevelElement
+    max_alarm: Not specified
+    max_dim_x: 1
+    max_dim_y: 0
+    max_value: Not specified
+    min_alarm: Not specified
+    min_value: Not specified
+    name: loggingLevelElement_B
+    standard_unit: No standard unit
+    unit: ''
+    writable: READ_WRITE
+    writable_attr_name: logginglevelelement
+  - data_format: SCALAR
+    data_type: DevLong
+    description: Current  logging level to logging target
+    disp_level: OPERATOR
+    display_unit: No display unit
+    format: '%d'
+    label: loggingLevel
+    max_alarm: Not specified
+    max_dim_x: 1
+    max_dim_y: 0
+    max_value: Not specified
+    min_alarm: Not specified
+    min_value: Not specified
+    name: OtherAttribute
+    standard_unit: No standard unit
+    unit: ''
+    writable: READ_WRITE
+    writable_attr_name: logginglevel
+  commands:
+  - disp_level: OPERATOR_B
+    doc_in: ON/OFF_B
+    doc_out: Uninitialised
+    dtype_in: DevBoolean
+    dtype_out: DevVoid
+    name: Capture
+  - disp_level: EXPERT
+    doc_in: Uninitialised
+    doc_out_B: Uninitialised
+    dtype_in: DevVoid
+    dtype_out: DevVoid
+    name: ClearOldTasks
+  - disp_level: EXPERT
+    doc_in: Uninitialised
+    doc_out: Uninitialised
+    dtype_in: DevVoid
+    dtype_out: DevVoid
+    name: ClearTaskHistory_B
+  - disp_level: EXPERT
+    doc_in: Uninitialised
+    doc_out: Uninitialised
+    dtype_in: DevVoid
+    dtype_out: DevVoid
+    name: OtherCommand
+  properties:
+  - name: AdminModeDefault
+  - name: AsynchCmdReplyNRetries
+  - name: AsynchCmdReplyTimeout
+  - name: CentralLoggerEnabledDefault
+  - name: ConfigureTaskTimeout
+  - name: ControlModeDefault_B
+"""
+
+
+def test_validate():
+    """Test various combinations"""
+
+    # If it's the same then no differences recorded
+    assert not compare_data(YAML_A, YAML_A, 1)
+
+    spec_yaml = YAML_A
+    dev_yaml = YAML_B
+    bi_directional_result = compare_data(spec_yaml, dev_yaml, 1)
+    single_direction_result = compare_data(spec_yaml, dev_yaml, 0)
+    command_result = (
```

Comment:
Let's rather rename this to `class_result`. I see you have `attr_result` and `prop_res` for some.